### PR TITLE
[Event Hubs Client] Idempotent Publishing Live Tests

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to introduce the Live tests for the idempotent publishing feature, fix an error in AMQP metadata names, and address some delayed feedback from the previous set of changes.  The Live tests are currently disabled as the feature is not yet available across the public cloud and requires a special private resource to test against.

**Note:** These API changes are part of a preview feature. The design and API surface were informally reviewed with the language architect but have not yet undergone formal review with the architecture board. I expect some changes, particularly around naming.

# Last Upstream Rebase

Thursday, September 24, 2:28pm (EDT)

# References and Related Issues

- [Idempotent Publishing Design](https://gist.github.com/jsquire/1cc4db6b3ca4ef13d26dc5315483555b)
- [Idempotent Publishing Requirements with Usage](https://gist.github.com/jsquire/0f3bc5701388fe03ccd027e86f1994f1)
- [Idempotent Publishing Support](https://github.com/Azure/azure-sdk-for-net/issues/https://github.com/Azure/azure-sdk-for-net/issues/14710) ([#13941](https://github.com/Azure/azure-sdk-for-net/issues/https://github.com/Azure/azure-sdk-for-net/issues/14710))

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs [hunk 1]
```diff
@@ -638,14 +636,15 @@ public void Dispose()
                     linkSettings.DesiredCapabilities.Add(AmqpProperty.EnableIdempotentPublishing);
                 }
 
-                // These values must either all be specified or none.  It is valid for an individual value to be null to signal the
-                // service to supply the value, so long as the key is present.
+                // The property keys on the link must either all be specified or none.  It is valid for an individual option to be missing, which
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs:639]: @christothes:  Updated this to address your feedback in the previous PR.   Does this better explain?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs [hunk 2]
```diff
@@ -401,7 +401,7 @@ public virtual async Task<string[]> GetPartitionIdsAsync(CancellationToken cance
         ///
         /// <returns>The set of information about the publishing state of the requested partition, within the context of this producer.</returns>
         ///
-        public virtual async Task<PartitionPublishingProperties> ReadPartitionPublishingPropertiesAsync(string partitionId,
+        public virtual async Task<PartitionPublishingProperties> GetPartitionPublishingPropertiesAsync(string partitionId,
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs:404]: nit:  Fix indentation

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+    ///   The suite of live tests for the idempotent publishing feature of the
+    ///   <see cref="EventHubProducerClientClient" /> class.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   These tests have a dependency on live Azure services and may
+    ///   incur costs for the associated Azure subscription.
+    /// </remarks>
+    ///
+
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs:23]: nit:  Fix blank

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+                var events = EventGenerator.CreateEvents(10).ToArray();
+                Assert.That(events.Any(item => item.PublishedSequenceNumber.HasValue), Is.False, "Events should start out as unpublished with no sequence number.");
+
+                await producer.SendAsync(events, sendOptions, cancellationSource.Token);
+                Assert.That(events.All(item => item.PublishedSequenceNumber.HasValue), Is.True, "Events should be sequenced after publishing.");
+
+                foreach (var item in events)
+                {
+                    Assert.That(item.PublishedSequenceNumber, Is.EqualTo(++eventSequenceNumber), $"The sequence numbers should be contiguous.  Event { eventSequenceNumber + 1 } was out of order.");
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs:324]: nit: I'm not sure I understand why the assert message specifies that `eventSequenceNumber + 1` is the one out of order. Wouldn't it be the current `eventSequenceNumber` as incremented in the `EqualTo` statement?

**Review by christothes — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs (hunk 4 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs:324]: d'oh.   Yup, that's a mistake.

**Pull request merged**